pkg/service/volume_service_test.go

```diff
@@ -101,6 +102,50 @@ func (suite *VolumeServiceTestSuite) TestGetNonExistentVolumeByID() {
 	}
 }
 
+func (suite *VolumeServiceTestSuite) TestGetVolumesPagination() {
+	arg := new(param.GetParams)
+	arg.Page = new(pagination.GetPage)
+
+	// set batch size
+	arg.Page.SetPageSize(2)
+	// fetch only 2 volumes
+	volumes, err := suite.volumeService.GetVolumes(arg)
+	if err != nil {
+		suite.T().Errorf("TestGetVolumesPagination(): Unable to fetch volumes, err: %v", err.Error())
+		return
+	}
+
+	if len(volumes) != 2 {
+		suite.T().Errorf("TestGetVolumesPagination(): pagination count doesnt match with # of return volumes ")
+		return
+	}
+
+	// Batch processing
+	for arg.Page.ContainsMore() {
+		volumes, err := suite.volumeService.GetVolumes(arg)
+		if err != nil {
+			suite.T().Errorf("TestGetVolumesPagination(): Unable to fetch volumes, err: %v", err.Error())
+			return
+		}
+		fmt.Printf("contains more volumes , parmas %+v, volumes : %v", arg, volumes)
+	}
+
+	// dont set batch size, should return all the volumes
```

Comment:
Should it be a separate test case/function?

Reply:
There is no strict rule since this is part of pagination so I kept it in a single function.

Reply:
Okay.  Generally I believe it is considered good practice to keep unit test cases small and targeted.  GoLang seems to prefer test cases defined in a structured list of inputs and outputs, though that might not be straightforward to apply here.